src/call.cpp

```diff
@@ -985,4 +985,46 @@ void DarlingServer::Call::S2CPerform::processCall() {
 	_sendReply(code);
 };
 
+void DarlingServer::Call::SetExecutablePath::processCall() {
+	int code = 0;
+	int fullLength = 0;
+
+	if (auto thread = _thread.lock()) {
+		if (auto process = thread->process()) {
+			std::string tmpstr;
+			tmpstr.resize(_body.buffer_size);
+			if (!process->readMemory((uintptr_t)_body.buffer, tmpstr.data(), _body.buffer_size, &code)) {
+				code = -code;
+			}
+			process->setExecutablePath(tmpstr.c_str());
```

Comment:
Two things:
  * Following the change to the signature of `setExecutablePath` in the header before, we can change this to just `tmpstr`.
  * You should only set the executable path if the memory read succeeded.
```suggestion
			} else {
				process->setExecutablePath(tmpstr);
			}
```